openclsim/core/events_container.py

```diff
@@ -15,34 +17,41 @@ class EventsContainer(simpy.FilterStore):
     """
 
     def __init__(self, env, store_capacity: int = 1, *args, **kwargs):
-        super().__init__(env, capacity=store_capacity)
+        super().__init__(env, capacity=store_capacity * 2)
         self._env = env
-        self._get_available_events = {}
-        self._put_available_events = {}
-
-    def initialize(self, init=0, capacity=0):
-        """Initialize method is a convenience method for backwards compatibility reasons."""
-        self.put(init, capacity)
+        self._container_events = {}
 
     def initialize_container(self, initials):
         """Initialize method used for MultiContainers."""
+
         for item in initials:
             assert "id" in item
             assert "capacity" in item
             assert "level" in item
-            super().put(item)
-
-    def get_available(self, amount, id_="default"):
-        if self.get_level(id_) >= amount:
-            return self._env.event().succeed()
-        if id_ in self._get_available_events:
-            if amount in self._get_available_events[id_]:
-                return self._get_available_events[id_][amount]
-        # else case: id_ is not in self._get_available_events
-        new_event = self._env.event()
-        self._get_available_events[id_] = {}
-        self._get_available_events[id_][amount] = new_event
-        return new_event
+            assert "_reservations" not in item["id"]
+
+            container_item = {
+                "id": item["id"],
+                "capacity": item["capacity"],
+                "level": item["level"],
+            }
+            reservation_item = {
+                "id": f"{item['id']}_reservations",
+                "capacity": item["capacity"],
+                "level": item["level"],
+            }
+
+            super().put(container_item)
+            super().put(reservation_item)
+
+    @property
+    def container_list(self):
+        container_ids = []
+        if len(self.items) > 0:
+            container_ids = [
+                item["id"] for item in self.items if "_reservations" not in item["id"]
```

Comment:
I don't know if the container ID is user-defined, but better safe than sorry: `not item["id"].endswith("_reservations")`

Also, if `self.item` is guaranteed to be a `list`, you don't need the `if`:

```python
return [
        item["id"] for item in self.items if not item["id"].endswith("_reservations")
]
```

Otherwise you need some other check, like: `if self.items:` or `self.items or []`.